Author: Florian Forster <octo@verplant.org>
Description:
 Undo upstream removal of automatic reading of the track lengths in TOC files
 (Closes: #440419).

Index: cdrdao-1.2.3/trackdb/TocParser.g
===================================================================
--- cdrdao-1.2.3.orig/trackdb/TocParser.g	2009-05-20 08:21:37.000000000 +0200
+++ cdrdao-1.2.3/trackdb/TocParser.g	2010-07-04 14:17:48.000000000 +0200
@@ -630,6 +630,15 @@ samples > [ unsigned long s ]
     (  msf > [ m ] << $s = m.samples(); >>
      | uLong > [ $s ]
     )
+    << if ($st != NULL && $st->length() == 0) {
+         // try to determine length 
+         if ($st->determineLength() != 0) {
+	   message(-2, "%s:%d: Cannot determine length of track data specification.",
+		   filename_, $lineNr);
+	   error_ = 1;
+	 }
+       }
+    >> 
     ;
     // fail action
     << $s = 0; >>
